public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx

```diff
@@ -151,26 +152,112 @@ export const ConnectionSVG = ({
       const CONNECTION_HEAD_ID_START = `connectionHeadStart-${headId + Math.random()}`;
       const CONNECTION_HEAD_ID_END = `connectionHeadEnd-${headId + Math.random()}`;
 
+      const radius = strokeRadius;
       // Create vertex path and populate array of add vertex controls
       const addVertices: ConnectionCoordinates[] = [];
       let pathString = `M${x1} ${y1} `;
       if (vertices?.length) {
         vertices.map((vertex, index) => {
           const x = vertex.x;
           const y = vertex.y;
-          pathString += `L${x * (x2 - x1) + x1} ${y * (y2 - y1) + y1} `;
+
+          const X = x * (x2 - x1) + x1;
```

Comment:
This whole section can be significantly simplified and made more robust to properly handle clockwise vs counter clockwise combinations.